File: Kotlin/Algorithm/src/차이를 최대로.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.abs
import kotlin.math.max

fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val arr = readLine().split(" ").map { it.toInt() }.toIntArray()
    val visited = BooleanArray(n)
    var answer = 0

    fun permutation(len: Int, result: IntArray, n: Int, arr:IntArray, visited:BooleanArray) {
        if(len == n) {
            var sum = 0
            for (i in 0 until n-1) {
                sum += abs(result[i]-result[i+1])
            }
            answer = max(answer, sum)
            return
        }

        for (i in 0 until n) {
            if (visited[i]) continue
            result[len] = arr[i]
            visited[i] = true
            permutation(len+1, result, n, arr, visited)
            visited[i] = false
        }
    }

    permutation(0, IntArray(n), n, arr, visited)
    println(answer)
}

File: python/백준 온라인 저지/2022.02.01/차이를 최대로.py
import sys
input = sys.stdin.readline

n = int(input())

arr = list(map(int, input().split()))

def permutation(arr, n):
    result = []

    if n == 0:
        return [[]]

    for i, ele in enumerate(arr):
        for P in permutation(arr[:i]+arr[i+1:], n-1):
            result += [[ele] + P]
    
    return result
    
answer = 0

for permu in permutation(arr, n):
    res = 0
    for i in range(n-1):
        res += abs(permu[i]-permu[i+1])
    
    if res > answer:
        answer = res

print(answer)